osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs

```diff
@@ -129,7 +131,7 @@ private void load()
                                         Origin = Anchor.CentreRight,
                                         Direction = FillDirection.Horizontal,
                                         Spacing = new Vector2(2),
-                                        Children = Score.Mods.Select(mod => new ModIcon(mod)
+                                        Children = Score.Mods.Select(mod => new ModIcon(rulesets.GetRuleset(Score.RulesetID).CreateInstance().CreateModFromAcronym(mod.Acronym))
```

Comment:
Maybe a smidge wasteful given that this will instantiate N instances of the ruleset for N mods rather than just 1. May be worth splitting out to a method for this?

Reply:
Weird, I thought I already split this out into a method.

Reply:
This is changed completely in https://github.com/ppy/osu/pull/15364, so it's probably not worth applying a change here.

The new version is just as inefficient, but I don't believe it's a huge issue.